compiler/x/codegen/FPTreeEvaluator.cpp

```diff
@@ -524,12 +524,18 @@ TR::Register *OMR::X86::TreeEvaluator::fpUnaryMaskEvaluator(TR::Node *node, TR::
       result->setIsSinglePrecision();
       }
 
-   // TODO 3-OP Optimization
-   if (result != value)
+   if (cg->comp()->target().cpu.supportsAVX())
       {
-      generateRegRegInstruction(TR::InstOpCode::MOVDQURegReg, node, result, value, cg);
+      generateRegRegMemInstruction(opcode, node, result, value, generateX86MemoryReference(cg->findOrCreate16ByteConstant(node, mask), cg), cg);
```

Comment:
To save some shared library code space, can you hoist the `generateX86MemoryReference(cg->findOrCreate16ByteConstant(node, mask), cg)`  above the `if` and share it on both branches?